girder/api/v1/resource.py

```diff
@@ -369,3 +374,55 @@ def copyResources(self, params):
         .errorResponse('No resources specified.')
         .errorResponse('Resource not found.')
         .errorResponse('ID was invalid.'))
+
+    @access.user
+    def mergeResources(self, params):
+        """
+        Merge the selected resources to a new parent folder. Only folder and
+        item resources can be moved with this function.
+        """
+        user, resources, parent, parentType, progress = \
+            self._prepareMoveCopyOrMerge(params, operation='merge')
+        name = params.get('name')
+        description = params.get('description', '')
+        total = sum([len(resources[key]) for key in resources])
+        with ProgressContext(progress, user=user, title='Moving resources',
+                             message='Calculating requirements...',
+                             total=total) as ctx:
+            for kind in resources:
+                model = self._getResourceModel(kind)
+                toBeMerged = {
+                    'item': [],
+                    'folder': []
+                }
+                for id in resources[kind]:
+                    doc = model.load(id=id, user=user, level=AccessType.WRITE)
+                    if not doc:
+                        raise RestException('Resource %s %s not found.' %
+                                            (kind, id))
+                    if kind in ('item', 'folder'):
+                        toBeMerged[kind].append(doc)
+            return self.model('item').merge(toBeMerged, user, parent, name,
+                                            description, ctx)
+
+    mergeResources.description = (
+        Description('Merge a set of items and folders.')
+        .param('resources', 'A JSON-encoded list of types to move.  Each type '
+               'is a list of ids.  Only folders and items may be specified.  '
+               'For example: {"item": [(item id 1), (item id2)], "folder": '
+               '[(folder id 1)]}.')
+        .param('parentType', 'Parent type for the new parent of these'
+               'resources.')
+        .param('parentId', 'Parent ID for the new parent of these resources.')
+        .param('name', 'The name of the new item to be created')
+        .param('description',
+               'The description of the new item to be created.',
+               required=False)
+        .param('progress', 'Whether to record progress on this task. Default '
+               'is false.', required=False, dataType='boolean')
+        .errorResponse('Unsupport or unknown resource type.')
+        .errorResponse('Invalid resources format.')
+        .errorResponse('Resource type not supported.')
```

Comment:
"Resource type not supported" and "Unsupported or unknown resource type" should probably just be one thing, or is there a difference?